server/src/storage/data_transfer.rs

```diff
@@ -109,13 +106,31 @@ impl DataTransfer {
 
     /// Insert the compressed file into the files to be transferred. Retrieve the size of the file
     /// and add it to the total size of the current local files under the key.
-    pub fn add_compressed_file(&mut self, key: &u64, file_path: &Path) -> Result<(), ParquetError> {
+    pub async fn add_compressed_file(
+        &mut self,
+        key: &u64,
+        file_path: &Path,
+    ) -> Result<(), ParquetError> {
         let file_size = file_path.metadata()?.len() as usize;
         *self.compressed_files.entry(*key).or_insert(0) += file_size;
 
         // If the combined size of the files is larger than the batch size, transfer the data to the remote object store.
         if self.compressed_files.get(key).unwrap() >= &self.transfer_batch_size_in_bytes {
-            self.transfer_data(key)?;
+            self.transfer_data(key).await?;
+        }
+
+        Ok(())
+    }
+
+    /// Transfer all compressed files currently in the data folder to the remote object store.
+    /// Return [`Ok`] if all files were transferred successfully, otherwise [`ParquetError`].
+    pub(crate) async fn flush_compressed_files(&mut self) -> Result<(), ParquetError> {
```

Comment:
Maybe add a comment regarding what to do if this method fails, e.g., is it OK to simply call the method again or must half-transferred files be removed from the object store first? Likewise for `transfer_data()`.

Reply:
I have added some minor text that explains that the flush function can be called again if it fails since the data is transferred for each key one at a time. However, I do not think we can add much text about the transfer data function failing since handling partial uploads should be down to the cloud provider. 

Text about how we handle deleting the files on the edge after transferring and how this works when the method fails might be added later when the functionality for that is added.

Reply:
From the documentation of [`object_store::ObjectStore`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html) it looks like only [`put_multipart()`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html#tymethod.put_multipart) needs to be manually cleaned up using [`abort_multipart()`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html#tymethod.abort_multipart) as a cleanup method does not seem to exist for [`put()`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html#tymethod.put). So the additional documentation you have added is probably fine for now.